lib/net.js

```diff
@@ -1735,11 +1735,8 @@ function Server(options, connectionListener) {
   if (typeof options.highWaterMark !== 'undefined') {
     validateNumber(
       options.highWaterMark, 'options.highWaterMark',
+      0
```

Comment:
I'm throwing an error if `options.highWaterMark` is less than zero. Alternatively, we can fallback to a default value of `undefined`.

Reply:
I'm okay with being more strict and forbid negative values.